packages/@lwc/ssr-runtime/src/reflection.ts

```diff
@@ -11,11 +11,27 @@ import {
     create,
     entries,
     hasOwnProperty,
+    htmlPropertyToAttribute,
     isNull,
     toString,
 } from '@lwc/shared';
 
 import type { LightningElement } from './lightning-element';
+import type { Attributes } from './types';
+
+/**
+ * Remove global HTML attributes that correspond to public API properties. If a
+ * component overrides a global HTML attribute we "disable" reflection through
+ * removal.
+ */
+export function deletePublicApiAttrs(attrs: Attributes, publicProps: Array<string>) {
+    publicProps.forEach((publicPropName) => {
```

Comment:
The property doesn't have to be public for the global html attribute to be overridden,